packages/compass-import-export/src/components/import-preview.tsx

```diff
@@ -47,22 +59,176 @@ const cellUncheckedStyles = css({
+  }
+
+  // once the user manually changed the type, make the warning go away
+  if (result.detected !== selectedType) {
+    return null;
+  }
+
+  return (
+    <Tooltip
+      align="top"
+      justify="middle"
+      delay={500}
+      trigger={({ children, ...props }) => (
+        <div {...props}>
+          {children}
+          <div className={infoIconCSS}>
+            <Icon glyph="InfoWithCircle"></Icon>
+          </div>
+        </div>
+      )}
+    >
+      <>
+        <Body as="p">
+          This field has{' '}
+          {selectedType === 'number'
+            ? 'mixed numeric types'
+            : 'mixed data types'}
+          :
+        </Body>
+        <ul className={typesListCSS}>
+          {Object.entries(result.types).map(([type, info]) => {
+            return (
+              <li key={type}>
+                {type} * {info.count}
+              </li>
+            );
+          })}
+        </ul>
+        <Body as="p">To standardise your data, select a different type.</Body>
+      </>
+    </Tooltip>
+  );
+}
+
+function FieldHeader(
+  field: Field,
+  analyzed: boolean,
+  onFieldCheckedChanged: (fieldPath: string, checked: boolean) => void,
+  setFieldType: (fieldPath: string, fieldType: string) => void
+) {
+  return (
+    <TableHeader
+      key={`col-${field.path}`}
+      label={
+        <div
+          className={columnHeaderStyles}
+          data-testid={`preview-field-header-${field.path}`}
+        >
+          {field.type !== 'placeholder' && (
+            <>
+              <div className={columnNameStyles}>
+                <Checkbox
+                  aria-labelledby={`toggle-import-field-label-${field.path}`}
+                  id={`toggle-import-field-checkbox-${field.path}`}
+                  data-testid={`toggle-import-field-checkbox-${field.path}`}
+                  aria-label={
+                    field.checked
+                      ? `${field.path} values will be imported`
+                      : `Values for ${field.path} will be ignored`
+                  }
+                  checked={field.checked}
+                  title={
+                    field.checked
+                      ? `${field.path} values will be imported`
+                      : `Values for ${field.path} will be ignored`
+                  }
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                    onFieldCheckedChanged(field.path, !!e.target.checked)
+                  }
+                />
+                <Label
+                  id={`toggle-import-field-label-${field.path}`}
+                  className={fieldPathHeaderStyles}
+                  htmlFor={`toggle-import-field-checkbox-${field.path}`}
+                >
+                  <span title={field.path}>{field.path}</span>
+                </Label>
+              </div>
+              <div className={fieldTypeContainerStyles}>
+                {analyzed ? (
+                  <>
+                    <SelectFieldType
```

Comment:
Small thing, feel free to ignore, and something I should have caught in the older import/export LeafyGreen work:
This `SelectFieldType` looks like a simple component wrapper around a `Select`. Do you feel it's nicer to have it all in this file and not a seperate `select-field-type.tsx`?